ERCS/erc-7390.md

```diff
@@ -0,0 +1,364 @@
+
+If she decides to sell the TokenA tokens, she'd make a profit of 8\*50 - 8\*25 = 200 TokenB!
+
+#### Put Option
+
+Let's say Bob sells a put option to Alice.\
+He gives the right to Alice to sell **8 TokenA** tokens at **25 TokenB** each between **14th of July 2023** and **16th of July 2023**.\
+For such a contract, he asks Alice to give him **10 TokenC** as a premium.\
+
+To create the contract, he will give the following parameters:
+
+- `side`: **Put**
+- `underlyingToken`: **TokenA's address**
+- `amount`: **8 \* 10^(TokenA's decimals)**
+- `strikeToken`: **TokenB's address**
+- `strike`: **8 \* 25 \* 10^(TokenB's decimals)**
+- `premiumToken`: **TokenC's address**
+- `premium`: **10 \* 10^(TokenC's decimals)**
+- `exerciseWindowStart`: **1689292800** *(2023-07-14 timestamp)*
+- `exerciseWindowEnd`: **1689465600** *(2023-07-16 timestamp)*
+
+Bob has to transfer collateral to the contract. This collateral corresponds to the funds he will have to give to Alice if she decides to exercise all the options. He has to give as collateral 200 TokenB (8 \* 25) and does that by calling the function `approve(address spender, uint256 amount)` on the TokenB's contract. As parameters he will give the contract's address (`spender`) and for `amount`: **(`strike`\*`amount` / 10^(TokenA's decimals))**. Then he can execute `create` on the contract for issuing the options.
+
+Alice for her part has to allow the spending of **10 TokenC** by calling `approve(address spender, uint256 amount)` on the TokenC's contract, with as parameters the contract's address (`spender`) and for `amount`: **10 \* 10^(TokenC's decimals)**. Then, she can execute `buy` on the contract in order to buy all the options.
+
+We're on the 15th of July, and Alice wants to exercise her options because 1 TokenA is traded at only 10 TokenB! To exercise she has to approve the transferring of **8 TokenA** tokens and call `exercise` on the contract. Bob receives 8 TokenA tokens, and Alice 200 TokenB (8 TokenA \* 25 TokenB). She just made a profit of 200 - 8\*10 = 120 TokenB!
+
+#### Retrieve collateral
+
+Let's say Alice never exercised his option because it wasn't profitable enough for her. To retrieve his collateral, Bob would have to wait for the current time to be greater than `exerciseWindowEnd`. In the examples, this characteristic is set to 2 days, so he would be able to get back his collateral from the 16th of July by simply calling `retrieveExpiredTokens`.
+
+## Rationale
+
+The proposed ERC option standard provides a simple yet powerful interface for options contracts on Ethereum. By standardizing the interface, it becomes easier for developers to build applications and platforms that support options trading, and users can seamlessly interact with different options contracts across multiple dApps.
```

Comment:
```suggestion
```

This paragraph is justifying the whole proposal; it answers the question "why does ERC-7390 need to exist?" This kind of content belongs in the motivation section.